### ISSUE TITLE
Build instructions

### ISSUE THREAD

**Issue by schoeberl** [leros-dev/leros-doc#1]
Instructions on how to build the compiler are too short. When I read the build.sh comments it looks like it should be called from within leros-llvm, but that fails with:

CMake Error at CMakeLists.txt:231 (message):
  In-source builds are not allowed.

  Please create a directory and run cmake from there, passing the path

  to this source directory as the last argument.

  This process created the file `CMakeCache.txt' and the directory
  `CMakeFiles'.

  Please delete them.

**Comment by mortbopet** [comment]
Addressed in [commit](https://github.com/leros-dev/leros-llvm/commit/53f2e5f7da7e76206df8f540e269545c5d801aec), seemed like the created build directory was (accidentally) named equally to the leros-llvm repository, thus creating an in-source build.

**Comment by schoeberl** [comment]
Still gives 2 errors:

-- Found assembler: /Users/martin/leros-dev/leros-llvm/../build-leros-llvm/bin/clang
CMake Error at CMakeLists.txt:8 (project):
  The CMAKE_C_COMPILER:

    /Users/martin/leros-dev/leros-llvm/../build-leros-llvm/bin/clang

  is not a full path to an existing compiler tool.

and some more

./build.sh: line 57: cd: lib: No such file or directory
./build.sh: line 58: cd: linux: No such file or directory
mv: rename libclang_rt.builtins-leros32.a to /Users/martin/leros-dev/leros-llvm/../build-leros-llvm/lib/clang/8.0.0/lib/libclang_rt.builtins-leros32.a: No such file or directory
./build.sh: line 63: cd: bin: No such file or directory
--2018-11-25 10:45:19--  https://github.com/leros-dev/leros-lib/blob/master/runtime/crt0.leros.o
Aufl"osen des Hostnamens github.com (github.com)... 192.30.255.112, 192.30.255.113
Verbindungsaufbau zu github.com (github.com)|192.30.255.112|:443 ... verbunden.
HTTP-Anforderung gesendet, auf Antwort wird gewartet ... 404 Not Found
2018-11-25 10:45:20 FEHLER 404: Not Found.

Is the build process run by executing ./build.sh within leros-llvm? 
Please write a note into the web site of leros-dev

BTW, why does leros-llvm not have issues?

**Comment by mortbopet** [comment]
[commit](https://github.com/leros-dev/leros-llvm/commit/0160e0d7b51ecda9d583d86242aa42c50c19e1c7) adress some issues in the build.sh script. It should be directory agnostic now - Ran it through to the end on my ubuntu machine, and everything works as intended.

Issues have now been enabled for leros-llvm repo.

**Comment by schoeberl** [comment]
I am still having troubles:

[100%] Built target llvm-xray
[100%] Linking CXX executable ../../bin/opt
[100%] Built target opt
Already up-to-date.
./build.sh: line 52: /Users/martin/leros-dev/build-leros-llvm/bin/clang: No such file or directory
mv: rename crt0.leros.o to /Users/martin/leros-dev/build-leros-llvm/lib/clang/8.0.0/crt0.leros.o: No such file or directory
build_runtime.sh: line 11: /Users/martin/leros-dev/build-leros-llvm/bin/clang: No such file or directory

**Comment by mortbopet** [comment]
Looks like the submodule targets (such as clang and lld) are not built on OSX (does successfully build on Linux). I will be back when i've investigated some more - i do not have access to an OSX machine so i cannot do any debugging as of the moment.

**Comment by schoeberl** [comment]
Sorry, but it does not build under Linux:

./build.sh: line 52: /home/martin/leros-dev/build-leros-llvm/bin/clang: No such file or directory
mv: cannot stat 'crt0.leros.o': No such file or directory

Did you try to do a build from scratch? Looks like there are dependencies not covered in the build.sh.

Could you please update either build.sh or the description how to build the complete compiler tool chain?

Thanks,
Martin

**Comment by mortbopet** [comment]
Working on it as we speak, issue didn't arise on my build machine for some reason (probably some PATH-related issues), but i have identified the issue.

**Comment by schoeberl** [comment]
It’s because clang is not built from build.sh. If you have it built (clang) and installed then sure the build.sh works. Try it from scratch (fresh checkout) without having any Leros clang installed.

Martin

**Comment by mortbopet** [comment]
The general issue was that build.sh did not check-out the submodules, so unless the user did a;
`git clone leros-llvm.git --recursive`  
clang and lld would not be built.  
I have modified the build script, and tested on a clean ubuntu machine, where it is working.
Regarding OSX, on OSX the cmake files for compiler-rt has some strange behaviour when the host system is an apple system, which needs to be resolved, llvm and clang compiles successfully on OSX, but not the runtime library.

For now, building through build.sh on linux seems to have been resolved (famous last words).

I have updated the build documentation in leros-doc.
